packages/editor-ui/src/constants.ts

```diff
@@ -2,7 +2,10 @@ import type { NodeCreatorOpenSource } from './Interface';
 import { NodeConnectionType } from 'n8n-workflow';
 
 export const MAX_WORKFLOW_SIZE = 16777216; // Workflow size limit in bytes
-export const MAX_WORKFLOW_PINNED_DATA_SIZE = 12582912; // Workflow pinned data size limit in bytes
+export const MAX_EXPECTED_REQUEST_SIZE = 2048; // Expected maximum workflow request metadata (i.e. headers) size in bytes
+export const MAX_PINNED_DATA_SIZE = import.meta.env.VUE_APP_MAX_PINNED_DATA_SIZE
+	? parseInt(import.meta.env.VUE_APP_MAX_PINNED_DATA_SIZE, 10)
```

Comment:
Should we guard against `NaN`?

Reply:
That would be overkill. We'll be the only ones using this in e2e tests anyway and this variable will remain undocumented.